File: algorithms/1_easy/C/03591_check_if_any_element_has_prime_frequency/solution.cpp
class Solution {
    inline bool is_prime(int num) {
        if (num == 2) {
            return true;
        }

        if (num <= 1 || num % 2 == 0) {
            return false;
        }

        int sqrt_num{static_cast<int>(sqrt(num))};

        for (int i{3}; i <= sqrt_num; i += 2) {
            if (num % i == 0) {
                return false;
            }
        }

        return true;
    }
public:
    bool checkPrimeFrequency(vector<int>& nums) {
        unordered_map<int, int> res{};

        for(const int &num: nums){
            res[num] += 1;
        }

        for (const auto &[val, freqs]: res) {
            if (is_prime(freqs)) {
                return true;
            }
        }
    
        return false;
    }
};

File: algorithms/1_easy/C/03591_check_if_any_element_has_prime_frequency/solution.py
class Solution:
    def checkPrimeFrequency(self, nums: List[int]) -> bool:
        res = {}
        for i in range(0,len(nums)):
            res[nums[i]] = res[nums[i]] + 1 if nums[i] in res else 1
        for v in res.values():
            if v > 1:
                is_prime = True
                for i in range(2,int(sqrt(v))+1):
                    if v % i == 0 :
                        is_prime = False
                        break
                if is_prime :
                    return True
        return False
